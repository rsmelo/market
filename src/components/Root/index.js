import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import router from '../../router'

const Root = ({ store }) => (
  <div>
    <Provider store={store}>
      <div>
        {router}
      </div>
    </Provider>
  </div>
)

Root.propTypes = {
  store: PropTypes.shape({}).isRequired,
}

export default Root
