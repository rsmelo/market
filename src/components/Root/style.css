.root {
  display: flex;
  flex-flow: wrap;
  justify-content: center;
}
