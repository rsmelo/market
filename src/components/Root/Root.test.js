import React from 'react'
import renderer from 'react-test-renderer'

import Root from './index'

jest.mock('react-redux', () => ({ Provider: 'Provider' }))
jest.mock('../../router', () => 'router')

describe('Root', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Root store={{}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
