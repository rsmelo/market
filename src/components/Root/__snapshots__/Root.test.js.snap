// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should render correctly 1`] = `
<Provider
  store={Object {}}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <h1>
          <a
            className={undefined}
            href="/"
            onClick={[Function]}
          >
            Market
          </a>
        </h1>
        <a
          href="/cart"
          onClick={[Function]}
        >
          <div
            className={undefined}
          />
        </a>
      </div>
    </div>
    router
  </div>
</Provider>
`;
